### ISSUE TITLE
RAMSES current_time discrepancy

### ISSUE THREAD

**Issue by cphyc** [yt-project/yt#2098]
### Bug report

**Bug summary**

When dealing with RAMSES data, the simulation time given by `ds.current_time` is not the same as the one you get from `ds.cosmology.t_from_z(ds.current_redshift)`.

**Code for reproduction**

```python
ds = yt.load('output_00080/info_00080.txt')
tA = ds.current_time.to('Myr')
tB = ds.cosmology.t_from_z(ds.current_redshift).to('Myr')

print(tA-tB)   # Returns -22.47323433951169 Myr
```
**Expected outcome**

I would expect to get exactly the same time(s). This is probably due to https://github.com/yt-project/yt/blob/c3aae45670d38281670fcfcd1196eb741cdf86aa/yt/frontends/ramses/io.py#L28.

**Comment by ngoldbaum** [comment]
Fair enough, it’s probably not necessary to have two implementations of the same thing. I think a PR making it use `t_from_z` would be accepted. Ping @markrichardson who added the function you’re pointing at.

**Comment by cphyc** [comment]
It may not be _as easy_ as this, because `convert_ramses_ages` takes as input a conformal time, while `t_from_z` expects a redshift as input. And we can't just throw `convert_ramses_ages` away since it is required to convert e.g. star ages from conformal to physical time. Any thoughts on this?

**Comment by ngoldbaum** [comment]
Does the cosmology calculator not have a way to go from conformal time to physical time? If it doesn't could we add it?

**Comment by cphyc** [comment]
This is actually a bit tricky: the conformal-to-physical time conversion requires to integrate the expression `dtau = H0 dt / a(t)^2`, where `H0` is the Hubble parameter and `a(t)` is the expansion factor. Most of the uncertainty seem to come from the integration part.
